packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -80,10 +87,14 @@ export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState
     let value = props.value || props.defaultValue || String(props.min) || '0';
     this._lastValidValue = value;
 
+    // Ensure that the autocalculated precision is not negative.
+    const precision = props.precision || Math.max(calculatePrecision(props.step || SpinButton.defaultProps.step!), 0);
```

Comment:
just do this.props.precision and `this.props.step` so you don't need to reference the default props directly. `this.props` will get set in super(props) so you can use it immediately afterwards and it will handle the default for you.